nntrainer/tensor/manager.cpp

```diff
@@ -431,6 +435,17 @@ std::vector<Weight *> Manager::requestWeights(
         grad = tensor_pool.requestOrExtend(shared_name + Var_Grad::grad_suffix,
                                            dim_g, grad_exec_order, grad_ls,
                                            Initializer::ZEROS);
+
+        if (var->getDataType() != ml::train::TensorDim::DataType::FP32) {
```

Comment:
It would be better to extract this part as a function or create a new flag later even if not now.

There are many confusing parts in development because we keep checking types and controlling execution flow in many files. 

If anyone has any good suggestions, it would be great if you could share them with us.

Reply:
As you know, the FP32 type variable is included in the weight for mixed precision. This should be assigned when the model requests the variable. But there is a special case like the batch normalization layer, which does not have FP16 var and grad because it needs to be computed with FP32. That means we do not need to allocate the Fp32 variable. Even if the mixed training mode is set in the model, we should check the type of the variable. 
But there are always better ways, so.. let's think about it.

Reply:
I got it ! Thank you for your detailed and kind answer